File: AspNetMvcUnleashed/0672329980_source/Chapter05/VB/EFMvcApplication/EFMvcApplication/Models/Repository.vb
Imports GenericRepository
Imports System.Data.Objects

Public Class Repository
    Inherits EFGenericRepository
    Implements IRepository

    Sub New(ByVal context As ObjectContext)
        MyBase.New(context)
    End Sub

    Public Function GetProductCount() As Integer Implements IRepository.GetProductCount
        Return Me.List(Of Product)().Count()
    End Function

End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter05/CS/EFMvcApplication/EFMvcApplication/Models/Repository.cs
using System.Data.Objects;
using System.Linq;
using GenericRepository;

namespace EFMvcApplication.Models
{
    public class Repository : EFGenericRepository, IRepository
    {
        public Repository(ObjectContext context) 
            : base(context) { }
        

        #region IRepository Members

        public int GetProductCount()
        {
            return this.List<Product>().Count();
        }

        #endregion
    }
}
